p2p/peer_set_test.go

```diff
@@ -30,16 +34,17 @@ func TestAddRemoveOne(t *testing.T) {
 		t.Errorf("Failed to add new peer and increment size")
 	}
 
-	peerSet.Remove(peer)
-	if peerSet.Has(peer.Key) {
-		t.Errorf("Failed to remove peer")
-	}
-	if peerSet.Size() != 0 {
-		t.Errorf("Failed to remove peer and decrement size")
+	// Invoke Remove more than once to ensure that the stored
+	// value is purged anyways i.e take the different paths.
+	for i := 0; i < 3; i++ {
```

Comment:
Why 3 times?  Wouldn't 2 be enough ? Also, should we test the different case of the peer being at the end of the list ?

Reply:
So 3 times was a value out of the air, nothing too particular to it.